aws/resource_aws_key_pair.go

```diff
@@ -32,19 +32,13 @@ func resourceAwsKeyPair() *schema.Resource {
 				Computed:      true,
 				ForceNew:      true,
 				ConflictsWith: []string{"key_name_prefix"},
+				ValidateFunc:  validateMaxLength(255),
 			},
 			"key_name_prefix": &schema.Schema{
-				Type:     schema.TypeString,
-				Optional: true,
-				ForceNew: true,
-				ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
-					value := v.(string)
-					if len(value) > 100 {
-						errors = append(errors, fmt.Errorf(
-							"%q cannot be longer than 100 characters, name is limited to 255", k))
-					}
-					return
-				},
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validateMaxLength(100),
```

Comment:
can you add a comment here about the 255 limit

Reply:
I would rather change it to `validateMaxLength(255 - 26)` but was unsure if `100` was there for some reason. What do you think?